releasenotes/notes/callback-for-NN-classifier-regressor-8e1d5c887335e6d9.yaml

```diff
@@ -0,0 +1,11 @@
+---
+features:
+  - |
+    The classes ``NeuralNetworkClassifier`` and ``NeuralNetworkRegressor`` has a new optional argument ``callback`` 
+    with default value ``None``. If a ``Callable`` is passed as the value of ``callback`` instead, it will be used 
+    to access the intermediate data during the optimization. Two parameter values are passed to the callback during 
+    each evaluation: the weights for the objective function and the computed objective value.
+    A new method ``get_objective`` is added to the class ``TrainableModel``, which is the parent class of 
+    ``NeuralNetworkClassifier`` and ``NeuralNetworkRegressor``, which invokes the ``ObjectiveFunction.objective``
+    and ``callback`` to return a custom ``objective`` function for the optimization.
```

Comment:
Let's keep `get_objective` private. For now it is not a part of the model interface.

Reply:
Rectified in https://github.com/Qiskit/qiskit-machine-learning/pull/151/commits/2a8dbf61e6052a005a73e100da658ed1b976a5fe